### PR TITLE
rgw/rgw_lua: s/boost::filesystem/std::filesystem/

### PR DESCRIPTION
**Pull request by tchaikov**
for less dependencies on 3rd party libraries, and to reduce the
compilation time.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_lua.cc [hunk 1]
```diff
@@ -196,10 +196,11 @@ int list_packages(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optiona
 
 int install_packages(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_yield y, packages_t& failed_packages, std::string& output) {
   // luarocks directory cleanup
-  boost::system::error_code ec;
+  std::error_code ec;
   const auto& luarocks_path = store->get_luarocks_path();
-  boost::filesystem::remove_all(luarocks_path, ec);
-  if (ec.value() != 0 && ec.value() != ENOENT) {
+  std::filesystem::remove_all(luarocks_path, ec);
```
Comment (yuvalif) [inline src/rgw/rgw_lua.cc:201]: why do you call `remove_all()` twice?

Comment (tchaikov) [reply src/rgw/rgw_lua.cc:201]: @yuvalif sorry, forgot to remove the first one when adding the second one. fixed and repushed. could you take another look?

**Review by yuvalif — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- remove all linkages of `Boost::filesystem` in rgw.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-07-24_05:22:39-rados-wip-kefu-testing-2021-07-24-1158-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-07-24_10:59:41-rados-wip-kefu-testing-2021-07-24-1507-distro-basic-smithi/

**Pull request merged**